sqlglot/dialects/snowflake.py

```diff
@@ -28,7 +28,9 @@
 
 
 # from https://docs.snowflake.com/en/sql-reference/functions/to_timestamp.html
-def _build_timestamp(name: str, kind: exp.DataType.Type) -> t.Callable[[t.List], exp.Func]:
+def _build_timestamp(
+    name: str, kind: exp.DataType.Type, safe: bool = False
+) -> t.Callable[[t.List], exp.Func]:
```

Comment:
Let's rename this to `_build_datetime` since it now handles various types (time, timestamp, date).